admin/init.sls

```diff
@@ -22,22 +22,37 @@ UTC:
     - mode: 644
     - source: salt://{{ tpldir }}/files/hosts
 
-sshkeys-dir:
-  file.directory:
-    - name: {{ root.home }}/.ssh
-    - user: {{ root.user }}
-    - group: {{ root.group }}
-    - mode: 700
-
-sshkeys:
+/etc/sshd_config:
   file.managed:
-    - name: {{ root.home }}/.ssh/authorized_keys
-    - user: {{ root.user }}
-    - group: {{ root.group }}
-    - mode: 600
-    - contents:
-      {% for ssh_user in admin.ssh_users %}
-      - {% include tpldir ~ '/files/ssh/' ~ ssh_user ~ '.pub' %}
-      {% endfor %}
-    - require:
-      - file: sshkeys-dir
+    - user: root
+    {% if grains['os'] == 'MacOS' %}
+    - group: wheel
+    {% elif grains['os'] == 'Ubuntu' %}
+    - group: root
+    {% endif %}
+    - mode: 644
+    - source: salt://{{ tpldir }}/files/sshd_config
+
+wheel:
+    group.present
+
+{% for ssh_user in admin.ssh_users %}
+{{ ssh_user }}:
+    group.present:
+        - members:
```

Comment:
We need to create the group first, and then the user, so this state shouldn't have a `members` key as that would be a circular dependency. The `gid_from_name` parameter in the `user.present` state will take care of this.

Reply:
FYI, since the `group.present` will be empty at that point, just set it to an empty list as so to get the YAML to parse correctly: https://docs.saltstack.com/en/2016.3/topics/troubleshooting/yaml_idiosyncrasies.html#yaml-does-not-like-double-short-decs